tutorials/xr/xr_action_map.rst

```diff
@@ -0,0 +1,246 @@
+
+  Ergo, there currently is no way to know with certainty, which controller the user is actually using.
+
+.. warning::
+  Finally, and this trips up a lot of people, the bindings aren't set in stone. It is fully allowed, and even expected, that an XR runtime allows a user to customise the bindings.
+
+  At the moment none of the XR runtimes offer this functionality though SteamVR has an existing UI from OpenVRs action map system that is still accessible. This is actively being worked on however.
+
+Our first controller binding
+----------------------------
+
+Lets setup our first controller binding and we'll do so for the Touch controller.
+
+Press "Add profile" and find the touch controller and add it, if it is not in the list it may already have been added:
+
+.. image:: img/xr_add_touch_controller.webp
+
+Our UI now shows a panel for the left controller, and for the right controller. It shows all the possible inputs and outputs of the controller in this UI. We can use the ``+`` next to each entry to bind it to an action:
+
+.. image:: img/xr_select_action.webp
+
+Lets finish our configuration:
+
+.. image:: img/xr_touch_completed.webp
+
+Each action is bound the the given input or output for both controllers to indicate that we support the action on either controller. The exception is the movement action which is bound only to the right hand controller. It is likely that we would want to use the left hand thumbstick for a different purpose, say a teleport function.
+
+In developing your game however you have to account for the possibility that the user changes the binding and binds the movement to the left hand thumbstick. 
+
+Also note that our shoot and grab boolean actions are linked to inputs of type ``float``. As mentioned before OpenXR will do conversions between the two, but do read the warning given on that subject earlier in this document.
+
+The simple controller
+---------------------
+
+The "Simple controller" is a generic controller that OpenXR offers as a fallback. However as becomes painfully clear after we apply our mapping:
+
+.. image:: img/xr_simple_controller.webp
+
+The simple controller is far too simple and falls short for anything but the simplest of VR games.
+
+This is why many XR runtimes only use it as a last resort and will attempt to use bindings from one of the more popular systems as a fallback first.
+
+.. note::
+  Due to the fallback not being very useful it is tempting to provide bindings for every controller supported by OpenXR. This is what the default action map does.
+
+  However it is the recommendation from the OpenXR Working Group that only bindings for controllers actually tested by the developer are setup. The XR runtimes are designed with this in mind and can do a better job at rebinding a provided binding, then a developer who takes a best stab at bindings without testing if this leads to a comfortable experience for the end user.
```

Comment:
Does the WG have a recommendation regarding the simple controller? Wouldn't they want that one to be included if possible?

Reply:
Yes they want it included but they recognize that it falls far short for most games and is practically unusable. I can't say what they plan to do about it as that is still being debated, but it is on their radar.

Reply:
Also note, from experimentation and looking at the SDKs I know the following:
- SteamVR allows driver developers to nominate a primary fallback which by default is the Touch controller, seeing Touch is probably the most prevalent controller that's a pretty safe bet
- WMR has paths to nearly all major controllers, so if you have either a Vive Wand, Touch or Index controller profile, they'll have mappings to MR and G2 controllers.
- Meta I don't know for sure, but seeing you'll be testing on a Quest if you want to deploy to their store it's a given you'll be providing a Touch profile.
- PICO we know falls back to simple controller and that has been a pain
- Not sure what Monado does but pretty sure they do more intelligent stuff as well

I'm hesitant to put that into the docs though as these things can change rapidly.

Reply:
Definitively reasonable to not go into detail here. I just wondered whether the simple controller would be useful to include for very primitive games (even though people generally don't test on the simple controller, for the lack of that controller being a real thing).

Reply:
It is, look at a game like Beat Saber, it could fully run with just specifying the simple controller. That is why I find it important to make people aware off its existence, but reality is that it just isn't good enough for most applications.

Reply:
I've expanded on this section a little, hopefully the intention is a bit clearer now.